account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,406 @@
+    def action_hello_world_connection(self):
+        if not self.hmrc_configuration:
+            raise exceptions.Warning("Please select hmrc configuration before continuing!")
+        # making sure that the endpoint is what we have used in the data file as the name can be changed anytime.
+        endpoint_record = self.env['ir.model.data'].search([('res_id','=',self.id), ('model', '=', 'mtd.hello_world')])
+
+        if endpoint_record.name == "mtd_hello_world_endpoint":
+            self.which_button_type_clicked = "helloworld"
+            self.path = "/hello/world"
+            version = self._json_command('version')
+            self._logger.info(
+                "Connection button Clicked - endpoint name {}, ".format(self.name) +
+                "redirect URL:- {}, Path url:- {}".format(self.hmrc_configuration.redirect_url, self.path)
+            )
+        elif endpoint_record.name == "mtd_hello_application_endpoint":
+            self.which_button_type_clicked = "application"
+            self.path = "/hello/application"
+            version = self._json_command('version')
+            self._logger.info(
+                "Connection button Clicked - endpoint name {},".format(self.name) +
+                "redirect URL:- {}, Path url:- {}".format(self.hmrc_configuration.redirect_url, self.path)
+            )
+        elif endpoint_record.name == "mtd_hello_user_endpoint":
+            self.which_button_type_clicked = "user"
+            self.path = "/hello/user"
+            self._logger.info(
+                "Connection button Clicked - endpoint name {}, ".format(self.name) +
+                "redirect URL:- {}, Path url:- {}".format(self.hmrc_configuration.redirect_url, self.path)
+            )
+            # search for token record for the API
+            token_record = self.env['mtd.api_tokens'].search([('api_id', '=', self.api_name.id)])
+            self._logger.info(
+                "Connection button Clicked - endpoint name {} , ".format(self.name) +
+                "and the api is :- {} ".format(self.api_name)
+            )
+            if not token_record.access_token and not token_record.refresh_token:
+                self._logger.info(
+                    "Connection button Clicked - endpoint name {0}, No access token ".format(self.name) +
+                    "found and no refresh_token found from the token record table."
+                )
+                api_tracker = self.env['mtd.api_request_tracker'].search([
+                    ('request_sent', '=', True),
+                    ('response_received', '=', False)
+                ])
+                self._logger.info(
+                    "Connection button Clicked - endpoint name {}, ".format(self.name) +
+                    "Checking to see if a request is in process"
+                )
+                time_10_mins_ago = (datetime.now() - timedelta(minutes=10))
+                format_time_10_mins_ago = time_10_mins_ago.strftime('%Y-%m-%d- %H:%M:%S')
```

Comment:
I suggest a noun-phrase instead of a verb-phrase here

```
formatted_time_10_mins_ago
```

Also is the trailing dash on `%d-` intentional?  If so I suggest a comment above saying why that hyphen is necessary.

Reply:
Also you probably mean `utcnow()` if you're comparing against a string that's coming out of the Odoo API.